implementations/app-rust/src/templates/types_rs.rs

```diff
@@ -93,48 +100,67 @@ pub enum {{detect_keyword (to_upper type)}} {
 {{#each methods}}
 // URI: "{{../uri}}" //
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct {{to_upper ../type}}Args{{to_upper name}} {
+pub struct {{remove_module_suffix (to_upper ../type) }}Args{{to_upper name}} {
     {{#each arguments}}
     {{serde_rename_if_case_mismatch name}}pub {{detect_keyword (to_lower name)}}: {{to_rust (to_graphql_type this)}},
     {{/each}}
 }
 
 {{/each}}
 #[derive(Clone)]
-pub struct {{detect_keyword (to_upper type)}} {
-    uri: Uri,
-    invoker: Arc<dyn Invoker>,
-    env: Option<Vec<u8>>
+pub struct {{remove_module_suffix (detect_keyword (to_upper type))}} {
+    pub uri: Uri,
+    pub invoker: Arc<dyn Invoker>,
+    pub env: Option<Vec<u8>>
 }
 
-impl {{detect_keyword (to_upper type)}} {
-    pub fn new(uri: Option<Uri>, invoker: Option<Arc<dyn Invoker>>, env: Option<Vec<u8>>) -> {{detect_keyword (to_upper type)}} {
+impl {{remove_module_suffix (detect_keyword (to_upper type))}} {
+    pub fn new(invoke_options: Option<InvokeOptions>) -> {{remove_module_suffix (detect_keyword (to_upper type))}} {
         let mut config = PolywrapClientConfig::new();
```

Comment:
I know this was here before and isn't a new addition from you, but I think we should be lazily instantiating the default PolywrapClientConfig and PolywrapClient. What do you think?

Reply:
yah that make sense; will do it